apps/src/templates/manageStudents/PasswordReset.jsx

```diff
@@ -40,39 +49,66 @@ class PasswordReset extends Component {
       }
     };
 
-    $.ajax({
-      url: `/dashboardapi/sections/${sectionId}/students/${studentId}`,
+    fetch(`/dashboardapi/sections/${sectionId}/students/${studentId}`, {
       method: 'PATCH',
-      contentType: 'application/json;charset=UTF-8',
-      data: JSON.stringify(dataToUpdate)
+      headers: {
+        'Content-Type': 'application/json;charset=UTF-8'
+      },
+      body: JSON.stringify(dataToUpdate),
+      credentials: 'include'
     })
-      .done(data => {
-        this.setState({
-          isResetting: false,
-          input: ''
-        });
-        firehoseClient.putRecord(
-          {
-            study: 'teacher-dashboard',
-            study_group: 'manage-students',
-            event: 'reset-secret',
-            data_json: JSON.stringify({
-              sectionId: sectionId,
-              studentId: studentId,
-              loginType: 'email'
-            })
-          },
-          {includeUserId: true}
-        );
+      .then(res => {
+        if (res.ok) {
+          this.setState({
+            isResetting: false,
+            input: ''
+          });
+          this.recordResetSecret();
+          this.hidePasswordLengthFailure();
+        } else if (res.status === 400) {
+          return res.text().then(text => {
+            throw new Error(text);
+          });
+        } else {
+          const err = new Error('HTTP status code: ' + res.status);
+          err.response = res;
+          err.status = res.status;
+          throw err;
+        }
       })
-      .fail((jqXhr, status) => {
+      .catch(err => {
+        try {
+          const errorObj = JSON.parse(err.message);
+          if (errorObj.errors.includes(PASSWORD_TOO_SHORT_ERROR_MESSAGE)) {
```

Comment:
I'm worried that this is a little fragile.  Is this error generated by the rails validation?  Will rails localize it depending on the user?

Are there other options you considered here?

Reply:
For visibility, James and I discussed this in the office, the error message is automatically generated by rails and will be internationalized. Since we only want to display this error for password length failures, I'm going to move the validation of length to the front-end instead of relying on the back-end to return a specific response. If the length that's enforced on the back-end changes we'll need to also update it on the front-end, but that'd be an issue with how I've implemented it here as well (we'd always display 6).